packages/astro/src/content/data-store.ts

```diff
@@ -183,12 +294,46 @@ export class DataStore {
 export interface ScopedDataStore {
 	get: (key: string) => DataEntry | undefined;
 	entries: () => Array<[id: string, DataEntry]>;
-	set: (key: string, data: Record<string, unknown>, body?: string, filePath?: string) => void;
+	/**
+	 * Adds a new entry to the store. If an entry with the same ID already exists,
+	 * it will be replaced.
```

Comment:
We need to document the returned value and explain what `true` and `false` mean. Especially `false`, where the `digest` seems involved.